pandas/tests/frame/test_alter_axes.py

```diff
@@ -10,262 +10,288 @@
 import numpy as np
 
 from pandas.compat import lrange, PY2
-from pandas import (DataFrame, Series, Index, MultiIndex,
-                    RangeIndex, date_range, IntervalIndex,
-                    to_datetime)
+from pandas import (DataFrame, Series, Index, MultiIndex, RangeIndex,
+                    IntervalIndex, DatetimeIndex, Categorical, cut,
+                    Timestamp, date_range, to_datetime)
 from pandas.core.dtypes.common import (
     is_object_dtype,
     is_categorical_dtype,
     is_interval_dtype)
-import pandas as pd
-
-from pandas.util.testing import assert_series_equal, assert_frame_equal
 
 import pandas.util.testing as tm
 
 from pandas.tests.frame.common import TestData
 
+key = lambda x: x.name
```

Comment:
they are "container"-types for testing, in the sense that -- applied to `df['A']` -- they give the correct type of container. They are to be thought of like `Series` or `Index`, but since the bare `MultiIndex()`- constructor does not take a vector, I wrote it as a lambda. Makes testing all allowed cases nicely parametrisable.

Reply:
this is obtuse, I can't tell what you are doing. if you need something create it where you need it.

Reply:
Refactored. Hope you like it better like this - the advantage is that it stays local and that the generated test names are understandable